packages/@react-aria/radio/src/useRadio.ts

```diff
@@ -80,15 +80,19 @@ export function useRadio(props: AriaRadioProps, state: RadioGroupState, ref: Ref
   let interactions = mergeProps(pressProps, focusableProps);
   let domProps = filterDOMProps(props, {labelable: true});
   let tabIndex: number | undefined = -1;
-  if (state.selectedValue != null) {
-    if (state.selectedValue === value) {
+  if (interactions.tabIndex) {
```

Comment:
what's this check for?

Reply:
Here we check whether the `tabIndex` has been set based on the `excludeFromTabOrder` prop that we pass to the `useFocusable` hook. If we use this prop, it has priority. Otherwise, we use the previous logic according to the regular rules. We can simply check the `excludeFromTabOrder` prop and return `-1` here. Let me know if there is anything else I need to change.

Reply:
here's a few more tests which should guide us on the solution more
```
  it('should unset the tabIndex for group disabled', () => {
    let {getByRole} = render(
      <RadioGroup isDisabled excludeFromTabOrder>
        <Label>Test</Label>
        <Radio value="a">A</Radio>
      </RadioGroup>
    );
    let radio = getByRole('radio');

    expect(radio).not.toHaveAttribute('tabindex');
  });

  it('should unset the tabIndex for radio disabled', () => {
    let {getByRole} = render(
      <RadioGroup excludeFromTabOrder>
        <Label>Test</Label>
        <Radio isDisabled value="a">A</Radio>
      </RadioGroup>
    );
    let radio = getByRole('radio');

    expect(radio).not.toHaveAttribute('tabindex');
  });

  it('should exclude the radio from tab order even if selected', () => {
    let {getByRole} = render(
      <RadioGroup excludeFromTabOrder value="a">
        <Label>Test</Label>
        <Radio value="a">A</Radio>
      </RadioGroup>
    );
    let radio = getByRole('radio');

    expect(radio).toHaveAttribute('tabindex', '-1');
  });
```